containers/run.sh

```
@@ -0,0 +1,199 @@
+if [ -z "$DETECTED_DISPLAY" ]; then
+    DETECTED_DISPLAY=$(ls /tmp/.X11-unix/X* 2>/dev/null | head -1 | sed 's|/tmp/.X11-unix/X|:|')
+fi
+[ -n "$DETECTED_DISPLAY" ] || fail "No X11 display found.
+  Run from a desktop terminal, or: DISPLAY=:0 ./containers/run.sh"
+ok "Display: $DETECTED_DISPLAY"
+
+# ── 3. X authority ─────────────────────────────────────────────────────────────
+DETECTED_XAUTH="${XAUTHORITY:-}"
+if [ -z "$DETECTED_XAUTH" ] || [ ! -f "$DETECTED_XAUTH" ]; then
+    [ -f "$HOME/.Xauthority" ] && DETECTED_XAUTH="$HOME/.Xauthority" || \
+    DETECTED_XAUTH=$(ps -eo args 2>/dev/null | grep -oP -- '-auth \K\S+' | head -1 || true)
```

Shellcheck (0.11.0)

[info] Consider using pgrep instead of grepping ps output.

(SC2009)